pkg/controller/config/config_controller.go

```diff
@@ -256,9 +264,21 @@ func (r *ReconcileConfig) replayData(ctx context.Context, w *watch.Set) error {
 		}
 
 		for i := range u.Items {
+			syncKey := r.syncMetricsCache.GetSyncKey(u.Items[i].GetNamespace(), u.Items[i].GetName())
+			defer r.syncMetricsCache.ReportSync(&syncc.Reporter{Ctx: context.TODO()})
```

Comment:
We should defer this only once. Looping over `defer` will cause `ReportSync()` to be called `len(u.Items)` times.